airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java

```diff
@@ -60,44 +61,49 @@ private boolean isEphemeralKeysAndPurgingStagingData(final JsonNode config, fina
-      return new AirbyteConnectionStatus()
-          .withStatus(AirbyteConnectionStatus.Status.FAILED)
-          .withMessage("Could not connect with provided configuration. \n" + e.getMessage());
-    } finally {
+      final S3DestinationConfig s3Config = getS3DestinationConfig(findS3Options(config));
+      final EncryptionConfig encryptionConfig =
+          config.has(UPLOADING_METHOD) ? EncryptionConfig.fromJson(config.get(UPLOADING_METHOD).get(JdbcUtils.ENCRYPTION_KEY)) : new NoEncryption();
+      if (isEphemeralKeysAndPurgingStagingData(config, encryptionConfig)) {
+        return new AirbyteConnectionStatus()
+            .withStatus(Status.FAILED)
+            .withMessage(
+                "You cannot use ephemeral keys and disable purging your staging data. This would produce S3 objects that you cannot decrypt.");
+      }
+      S3BaseChecks.attemptS3WriteAndDelete(new S3StorageOperations(new RedshiftSQLNameTransformer(), s3Config.getS3Client(), s3Config), s3Config,
+          s3Config.getBucketPath());
+
+      final NamingConventionTransformer nameTransformer = getNamingResolver();
+      final RedshiftS3StagingSqlOperations redshiftS3StagingSqlOperations =
+          new RedshiftS3StagingSqlOperations(nameTransformer, s3Config.getS3Client(), s3Config, encryptionConfig);
+      final DataSource dataSource = getDataSource(config);
       try {
-        DataSourceFactory.close(dataSource);
+        final JdbcDatabase database = new DefaultJdbcDatabase(dataSource);
+        final String outputSchema = super.getNamingResolver().getIdentifier(config.get(JdbcUtils.SCHEMA_KEY).asText());
+        attemptSQLCreateAndDropTableOperations(outputSchema, database, nameTransformer, redshiftS3StagingSqlOperations);
+        return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED);
+      } catch (final ConnectionErrorException e) {
+        final String message = getErrorMessage(e.getStateCode(), e.getErrorCode(), e.getExceptionMessage(), e);
+        AirbyteTraceMessageUtility.emitConfigErrorTrace(e, message);
+        return new AirbyteConnectionStatus()
+            .withStatus(AirbyteConnectionStatus.Status.FAILED)
+            .withMessage(message);
       } catch (final Exception e) {
-        LOGGER.warn("Unable to close data source.", e);
+        LOGGER.error("Exception while checking connection: ", e);
+        return new AirbyteConnectionStatus()
+            .withStatus(AirbyteConnectionStatus.Status.FAILED)
+            .withMessage("Could not connect with provided configuration. \n" + e.getMessage());
+      } finally {
+        try {
+          DataSourceFactory.close(dataSource);
+        } catch (final Exception e) {
+          LOGGER.warn("Unable to close data source.", e);
+        }
       }
+    } catch (final Exception e) {
+      LOGGER.error("Check failed.", e);
+      throw new ConfigErrorException(e.getMessage() != null ? e.getMessage() : e.toString());
```

Comment:
what does the error message look like? (just want to check that it's something a user could reasonably understand + figure out how to fix)